src/lib/maplibre/utils.ts

```diff
@@ -11,21 +11,22 @@ import type {
 import { appVersion, privateResourceBaseUrl } from "lib/common";
 import type {
   DataDrivenPropertyValueSpecification,
-  FilterSpecification,
+  ExpressionSpecification,
   LayerSpecification,
   Map,
   StyleSpecification,
 } from "maplibre-gl";
+import { getBeforeId } from "./zorder";
 
 // Some methods take optional params. It's an error to pass in null or undefined, so use default values from
 // https://github.com/maplibre/maplibre-style-spec/blob/main/src/reference/v8.json.
 const defaultColor = "#000000";
 const defaultFilter = true;
 const defaultOpacity = 1;
 
-export const isPolygon: FilterSpecification = ["==", "$type", "Polygon"];
-export const isLine: FilterSpecification = ["==", "$type", "LineString"];
-export const isPoint: FilterSpecification = ["==", "$type", "Point"];
+export const isPolygon: ExpressionSpecification = ["==", "$type", "Polygon"];
+export const isLine: ExpressionSpecification = ["==", "$type", "LineString"];
+export const isPoint: ExpressionSpecification = ["==", "$type", "Point"];
 
 // This sets up a GeoJSON source. MapLibre's API isn't idempotent; you can't
```

Comment:
Note that not all these old helpers are gone yet, because there are still a few places in the codebase where sources/layers are managed by `.ts` files and not Svelte components.

Reply:
If you're commenting on the lines above (`isPolygon`), then yes, they're part of filter expressions -- just little helpers. If you mean below (`overwriteSource`, `overwritePolygonLayer`, etc), then they're the old way of defining sources and layers before this PR. There are still a few of the draw tools using them because they're `.ts`. But I'd maybe like to revisit how they're written and take advantage of svelte reactivity, now that I know it a bit better